src/common/AdaptiveStream.cpp

```diff
@@ -720,15 +722,32 @@ bool AdaptiveStream::start_stream(uint64_t startPts)
             current_rep_->get_segment(static_cast<size_t>(segmentId - current_rep_->GetStartNumber()));
       }
     }
+    else if (stream_changed_) // switching streams, align new stream segment no.
```

Comment:
I have lost most of the time due to this method AdaptiveStream::start_stream
because i was not aware that the AdaptiveStream::start_stream method is used to adjust/set the "current segment" and so all my changes that i was doing on ensure_segment always was failing.
After discovered this, i realized that was missing set "current segment" for a stream switching case